### PR TITLE
Student application

### PR DESCRIPTION
**Pull request by zsp2**

### CODE REVIEW

File: TriviaNation.Admin/ViewModels/GameCreationViewModel.cs [hunk 1]
```diff
@@ -3,6 +3,11 @@
 using System.Collections.ObjectModel;
 using System.Linq;
 using System.Windows;
+<<<<<<< HEAD
```
Comment (rme9) [inline TriviaNation.Admin/ViewModels/GameCreationViewModel.cs]: This needs to be fixed.

File: TriviaNation.Student/App.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		}
+
+		public static string LoginMessage { get; set; }
+
+		public static void OnLogout()
+		{
+			Application.Current.Properties.Remove("LoggedInUserId");
+			Application.Current.Properties.Remove("LoggedInUserName");
+
+			Application.Current.Shutdown();
```
Comment (rme9) [inline TriviaNation.Student/App.xaml.cs:68]: Do we want the entire application to close on logout, or do we just want to return to the login screen?

File: TriviaNation.Student/MainWindow.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+			InitializeComponent();
+
+			ShowLoginWindow();
+		}
+
+		private void ShowLoginWindow()
+		{
+			//this.Hide();
+
+			var loginvm = new LoginPopupViewModel();
```
Comment (rme9) [inline TriviaNation.Student/MainWindow.xaml.cs]: Go ahead and cleanup some of this commented out code.

File: TriviaNation.Student/Models/Territories.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TriviaNation.Student.Models
+{
+    class Territories
+    {
+        private string Name;
+        private string ControllerName = null;
```
Comment (rme9) [inline TriviaNation.Student/Models/Territories.cs]: Make all these auto-properties (add a get and set)

Comment (rme9) [reply TriviaNation.Student/Models/Territories.cs]: The deserializer needs the setter.

File: TriviaNation.Student/Util/RelayCommand.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Input;
+
+namespace TriviaNation.Util
+{
+	public class RelayCommand : ICommand
+	{
+		private readonly Action<object> _Execute;
+
```
Comment (rme9) [inline TriviaNation.Student/Util/RelayCommand.cs:13]: We should probably move this to shared code, but I can handle that in my branch.

File: TriviaNation.Student/ViewModels/GameBoardViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+using TriviaNation.Util;
+using TriviaNation.ViewModels;
+
+namespace TriviaNation.Student.ViewModels
+{
+    public class GameBoardViewModel : IViewModel
+    {
+        public GameSession Session = null;
+        //public string LoggedInUserName
+        //{
```
Comment (rme9) [inline TriviaNation.Student/ViewModels/GameBoardViewModel.cs]: Cleanup the commented out code.

File: TriviaNation.Student/ViewModels/GameSessionListViewModel.cs [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        public async void ActiveGameSessions()
+        {
+            using (var gdrive = new GameDriver())
+            {
+                var result = await gdrive.GetGameSessionsByStudent(Student.Email);
+                _Sessions = result;
+            }
+
+            OnPropertyChanged(nameof(Sessions));
+            //Sessions.Add(result);
```
Comment (rme9) [inline TriviaNation.Student/ViewModels/GameSessionListViewModel.cs]: Should this still be here? If it's not needed, delete it.

File: TriviaNation.Student/ViewModels/LoginPopupViewModel.cs [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+using TriviaNation.Student.ViewModels;
+using TriviaNation.Util;
+
+namespace TriviaNation.ViewModels
+{
+	public class LoginPopupViewModel : ViewModel
+	{
+        private StudentUser Student;
+        //private string name;
+		#region Login Message
```
Comment (rme9) [inline TriviaNation.Student/ViewModels/LoginPopupViewModel.cs]: cleanup

File: TriviaNation.Student/Views/GameSessionListView.xaml.cs [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+        //public List<string> Sessions = new List<string>();
+        public GameSessionListView(GameSessionListViewModel model)
+        {
+            InitializeComponent();
+            DataContext = model;
+            gsvm = model;
+
+            CurrentName.Content = model.Student.Name;
+            //this.Show();
+            //if (model?.Sessions?.Count > 0)
```
Comment (rme9) [inline TriviaNation.Student/Views/GameSessionListView.xaml.cs]: Clean this up

File: TriviaNation.Student/App.xaml.cs (hunk 2 above)
Comment (zsp2) [reply TriviaNation.Student/App.xaml.cs:68]: I'm fine with this closing on logout for now. We can implement returning to the log in screen later.

Comment (rme9) [reply TriviaNation.Student/App.xaml.cs:68]: Sounds good to me.

**Review by rme9 — APPROVED** [review]

**Pull request merged**